### ISSUE TITLE
Overriding quickmarks does not work

### ISSUE THREAD

**Issue by edi9999** [qutebrowser/qutebrowser#1591]
```
go to duckduckgo.com
press m 
press a
press <enter>

go to google.com
press m
press a 
press <enter>
# qutebrowser asks whether to overwrite quickmark
press <enter>

press O
```

Expected result : have only one quickmark (to google.com)
Actual result :  qutebrowser stores the two quickmarks after hitting O

**Comment by The-Compiler** [comment]
Reproduced - seems like this is only a bug in the `:open` completion.

**Comment by rcorre** [comment]
just noticed this. Probably a missing signal/slot wiring. Should be resolved once #2295 removes caching.

**Comment by The-Compiler** [comment]
Fixed by #2295.
